eis_toolkit/surface_attributes/parameters.py

```diff
@@ -0,0 +1,213 @@
+        return np.pi + np.arctan2(p, q)
+
+
+def _second_order_basic_set(
+    parameter: Sequence[str],
+    coefficients: tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray, np.ndarray],
+) -> np.ndarray:
+    p, q, r, s, t = coefficients
+
+    if parameter == "planc":
+        return _divide(-2 * (q**2 * r - p * q * s + p**2 * t), np.sqrt(np.power(p**2 + q**2, 3)))
+    elif parameter == "profc":
+        return _divide(
+            -2 * (p**2 * r + p * q * s + q**2 * t), (p**2 + q**2) * (np.sqrt(np.power(1 + p**2 + q**2, 3)))
+        )
+    elif parameter == "profc_min":
+        return -r - t - np.sqrt(np.square(r - t) + s**2)
+    elif parameter == "profc_max":
+        return -r - t + np.sqrt(np.square(r - t) + s**2)
+    elif parameter == "longc":
+        return _divide(-2 * (p**2 * r + p * q * s + q**2 * t), p**2 + q**2)
+    elif parameter == "crosc":
+        return _divide(-2 * (q**2 * r - p * q * s + p**2 * t), p**2 + q**2)
+    elif parameter == "rot":
+        return _divide((p**2 - q**2) * s - p * q * (r - t), np.sqrt(np.power(p**2 + q**2, 3)))
+    elif parameter == "K":
+        return (r * t - s**2) / np.square(1 + p**2 + q**2)
+    elif parameter == "genc":
+        return -2 * (r + t)
+    elif parameter == "tangc":
+        return _divide(-2 * (q**2 * r - p * q * s + p**2 * t), (p**2 + q**2) * np.sqrt(1 + p**2 + q**2))
+
+
+@beartype
+def first_order(
+    raster: rasterio.io.DatasetReader,
+    parameters: Sequence[Literal["G", "A"]],
+    scaling_factor: Optional[Number] = 1,
+    slope_tolerance: Optional[Number] = 0,
+    slope_gradient_unit: Literal["degrees", "radians", "rise"] = "radians",
+    slope_direction_unit: Literal["degrees", "radians"] = "radians",
+    method: Literal["Horn", "Evans", "Young", "Zevenbergen"] = "Horn",
+) -> dict:
+    """Calculates the first order surface attributes.
+
+    For compatibility for slope and aspect calculations with ArcGIS or QGIS, choose Method Horn (1981).
+
+    Returns:
+        Selected surface attributes and respective updated metadata.
+
+    Raises:
+        InvalidRasterBandException: Raster has more than one band.
+        NonSquarePixelSizeException: Pixel dimensions do not have same length.
+        InvalidParameterValueException: Wrong input parameters provided.
+    """
+    if raster.count > 1:
+        raise InvalidRasterBandException("Only one-band raster supported.")
+
+    if check_quadratic_pixels(raster) is False:
+        raise NonSquarePixelSizeException("Processing requires quadratic pixel dimensions.")
+
+    if scaling_factor <= 0:
+        raise InvalidParameterValueException("Value must be greater than 0.")
+
+    raster_array = raster.read()
+    raster_array = reduce_ndim(raster_array)
+    raster_array = nodata_to_nan(raster_array, nodata_value=raster.nodata)
+    raster_array = scale_raster(raster_array, scaling_factor)
+
+    cellsize = raster.res[0]
+    p, q, *NoneValues = coefficients(raster_array, cellsize, method)
```

Comment:
```suggestion
    p, q, *_= coefficients(raster_array, cellsize, method)
```

Underscore is used for collecting values that can be discarded.